server/src-lib/Hasura/RQL/DDL/Metadata.hs

```diff
@@ -405,7 +406,7 @@ fetchMetadataFromHdbTables defaultSourceConfig = liftTx do
 
   let tableMetadatas = mapFromL _tmTable $ HM.elems postRelMap
       sources = HM.singleton defaultSource $
-                SourceMetadata defaultSource tableMetadatas functions defaultSourceConfig
+                SourceMetadata defaultSource tableMetadatas functions defaultSourceConfig [] {- TODO: not sure what to do here -}
```

Comment:
I do think there's no notion of `read_replicas` in the OSS server before. This line of code is trying to migrate the metadata of the older single source database to the newer multi source support. Hence we can safely use `[]` which specifies there's no read_replica configuration.